numpy/core/src/multiarray/dtype_transfer.c

```diff
@@ -4625,6 +4625,7 @@ PyArray_GetMaskedDTypeTransferFunction(int aligned,
     /* TODO: Special case some important cases so they're fast */
 
     /* Fall back to wrapping a non-masked transfer function */
+    assert(dst_dtype != NULL);
```

Comment:
It does not make sense to ask for a decref function with a masked operation. So I thought an assert here makes some sense.

I have not yet added a test for `nditer.copy()` in this case, but it probably should go elsewhere, so might as well do this PR without that test.

Reply:
Sorry, the point was, this should add coverage to: `_strided_masked_wrapper_decsrcref_transfer_function` in `dtype_transfer.c`